activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -143,6 +149,10 @@ def collector
         end
       end
 
+      def arel_visitor # :nodoc:
+        (Arel::Visitors::VISITORS[@config[:adapter]] || Arel::Visitors::ToSql).new(self)
```

Comment:
Why fallback to `Arel::Visitors::ToSql`?
I think it is better to override this method if someone create an adapter that Arel does not support.